gremlin-go/driver/bytecode.go

```diff
@@ -152,8 +164,59 @@ type instruction struct {
 	arguments []interface{}
 }
 
-// TODO: AN-1018 Export this
-type binding struct {
-	key   string
-	value interface{}
+// Binding associates a string variable with a value
+type Binding struct {
+	Key   string
+	Value interface{}
+}
+
+// String returns the key value binding in string format
+func (b *Binding) String() string {
+	return fmt.Sprintf("binding[%v=%v]", b.Key, b.Value)
+}
+
+// Equals check for equality of current binding object to another
+func (b *Binding) Equals(item interface{}) bool {
+	if reflect.DeepEqual(b, item) {
+		return true
+	}
+	if (item == nil) || fmt.Sprintf("%T", b) != fmt.Sprintf("%T", item) {
+		return false
+	}
+	val, ok := item.(*Binding)
+	return ok && (b.Key == val.Key && reflect.DeepEqual(b.Value, val.Value))
+}
+
+// Hash returns the hashcode of the binding in int
+func (b *Binding) Hash() int {
+	h := fnv.New64a()
+	keyHash, err := h.Write([]byte(b.Key))
+	if err != nil {
+		return 0
+	}
+	var buff bytes.Buffer
+	enc := gob.NewEncoder(&buff)
+	err = enc.Encode(b.Value)
+	if err != nil {
+		return 0
+	}
+	valHash, err := h.Write(buff.Bytes())
+	if err != nil {
+		return 0
+	}
+	return keyHash + valHash
+}
```

Comment:
Remove Equals and Hash. This is python junk.
```suggestion
```

Reply:
Will remove locally